Notify about attestation state using LEDs

arturkow2000 opened the pull request.

arturkow2000 committed:
Implement LED driver

arturkow2000 committed:
Signal attestation status using LED

arturkow2000 committed:
Add dummy LED driver to pal_pc

This is needed to keep PAL interface uniform

krystian-hebel commented on src/client/attestation/state.rs:
```diff
@@ -35,13 +34,22 @@ pub enum State {
 
 impl State {
     /// Transition into error state.
     pub fn error(&mut self) {
```
Something similar should also be done for provisioning.

krystian-hebel replied:
Maybe we should also report success of provisioning by shorter/longer/blinking green LED?

arturkow2000 replied:
@krystian-hebel Done, when provisioning succeeds/fails Fobnails gives 3 quick blinks.
Also I reduced default retry limit from infinite to 5 (with timeout of 1 second) so that Fobnail signals failure after 5 of no response from attester.

krystian-hebel reviewed (changes requested):
Code works for attestation, but doesn't report failure during provisioning.

arturkow2000 committed:
Signal provisioning status using LED

Blink 3 times either with green or red LED depending on whether
provisioning is successful or not.

arturkow2000 committed:
Reduce default retry count in CoAP client

By default CoAP client had infinite number of retries, this caused
Fobnail to not give any signal when attester is not responding. Reduce
retry count to 5 so that Fobnail signals failure after 5 seconds of no
response.

arturkow2000 committed:
Fix misleading error message

We don't retry attestation on error anymore.
